qiskit/circuit/quantumcircuit.py

```diff
@@ -2096,10 +2123,13 @@ def copy(self, name: str | None = None) -> "QuantumCircuit":
             }
         )
 
-        cpy._data = [
-            instruction.replace(operation=operation_copies[id(instruction.operation)])
-            for instruction in self._data
-        ]
+        cpy._data.reserve(len(self._data))
+        cpy._data.extend(
+            (
+                instruction.replace(operation=operation_copies[id(instruction.operation)])
+                for instruction in self._data
+            )
+        )
```

Comment:
I'm kind of surprised that neither `black` nor `pylint` complained about the inner set of brackets here, since I don't think they're required by anything.

Reply:
Yeah, odd. Resolved in a39fe805dd0dac2c44e9bbf0704ac6b8c67e7eb1.